src/fluvio.test.ts

```diff
@@ -124,6 +125,58 @@ describe('Fluvio Batch Producer', () => {
     })
 })
 
+describe('Configures a SmartModule with a Filter configuration', () => {
+    jest.setTimeout(100000) // 100 seconds
+    let admin: FluvioAdmin
+    let fluvio: Fluvio
+    let topic: string
+
+    beforeAll(async () => {
+        topic = uuidV4()
+        fluvio = await Fluvio.connect()
+        admin = await fluvio.admin()
+
+        console.log(`Creating topic ${topic}`)
+
+        await admin.createTopic(topic)
+        await sleep(topic_create_timeout)
+    });
+
+    afterAll(async () => {
+        console.log(`Deleting topic ${topic}`)
+        await admin.deleteTopic(topic)
+        await sleep(topic_create_timeout)
+    });
+
+    test('Applies a Filter SmartModule on the provided stream', async () => {
+        const producer = await fluvio.topicProducer(topic)
+        const consumer = await fluvio.partitionConsumer(topic, 0);
+        const wasmSmartModule = await fs.promises.readFile('./fixtures/server_logs_filter.wasm.gz');
```

Comment:
This file is gziped, I think that from file we should support reading the file without compression and then compress that before sending data to fluvio

Reply:
Sounds good to me! I could give a shot on that, perhaps in a separate PR to keep things tidy/focused on each concern?